### PR TITLE
Fix: error messages for non-owners

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves

Resolves https://github.com/safe-global/web-core/issues/769, https://github.com/safe-global/web-core/issues/722

Contains #860 by @niklasnatter + my additions to handle the new tx case and gas limit errors.

## How this PR fixes it

Builds on top of @niklasnatter's PR and adjusts the owner condition:

```
const cannotPropose = !isOwner && !onlyExecute
```

Non-owners can only submit if execution is the only action. If a tx can be either signed or executed, it means it's not fully signed and thus cannot be executed by non-owners.

## How to test it
 It should show an error in the following cases:
* Create a tx and switch the wallet to a non-owner
* Start signing an existing tx and switch the wallet to a non-owner

It should NOT show an error when:
* Switch to a non-owner and try executing an existing fully signed tax

## Screenshots

Screenshot 2022-10-08 at 08 36 26

### CODE REVIEW

**Comment by francovenica** [comment]
The "wrong network" message error and the "non-owner error message" when creating the tx work fine.

Issue:
Trying to execute a tx that is in the queue list ready to be executed with a non-owner will trigger the "non-owner" error message.

With an owner, create a tx in a safe 1/x and queue it
Switch to a non-owner and try to execute the tx.
image

**Comment by katspaugh** [comment]
@francovenica on your screenshot, it's failing because it cannot estimate gas, there's no non-owner error displayed.

File: src/components/tx/SignOrExecuteForm/index.tsx [hunk 1]
```diff
@@ -166,7 +165,10 @@ const SignOrExecuteForm = ({
     setAdvancedParams(data)
   }
 
-  const submitDisabled = !isSubmittable || isEstimating || !tx || disableSubmit
+  const cannotPropose = !isOwner && !onlyExecute // Can't sign or create a tx if not an owner
+  console.log('isOwner', isOwner, 'onlyExecute', onlyExecute, 'cannotPropose', cannotPropose)
```
Comment (iamacook) [inline src/components/tx/SignOrExecuteForm/index.tsx:169]: 👀

**Review by iamacook — APPROVED** [review]
The link is a bit larger than the error text. Can you make it the same size?

image

**Comment by francovenica** [comment]
@katspaugh The issue seems to only happen with non-owners tho. If you try to execute with an owner the issue is not there. 
If you think this issue is unrelated to this ticket I can open a new one.

This is the safe https://error-messages.web-core.pages.dev/gor:0x0Bf3C9DA222Fb3e77Ea66C5D04130d35F6eEf9F4/transactions/tx?id=multisig_0x0Bf3C9DA222Fb3e77Ea66C5D04130d35F6eEf9F4_0xf033b4511064a358a2a9d2a391571da7937c977aac2a4d6406f73840b637cefb

**Comment by katspaugh** [comment]
Yes, it's unrelated, it's something with the gas estimation for non-owners. This PR merely displays warnings.

**Comment by francovenica** [comment]
Ok, I'll create a new ticket for it and close https://github.com/safe-global/web-core/issues/769, https://github.com/safe-global/web-core/issues/722

**Pull request merged**